### ISSUE TITLE
Add a repeating function that interrupts the loop

### ISSUE THREAD

**Issue by jusw85** [peoro/ngu.js#6]
In my branch, I set up a [function](https://github.com/jusw85/ngu.js/blob/dev/src/loops.js#L93-L145) that runs on an interval for setting up loadout, diggers, etc. before autokilling titans. 

When I open the game, I'd first run a killAllFixInv loop. Then because there's no OCR, I'd read off the remaining time till titan spawn and run a function directly in the console like `nguJs.loops.fn("0:28:40");`. This function will wait (initialDelay - 30s), stop the killAll loop, setup drop chance loadout + diggers(or gold drop if first kill in rebirth), then wait for autokill. Then it'd switch back to farming loadout, wait some time for E/M to recover. Then finally recap wandoos, magic, NGUs, and restart the killAll loop, then waits (titanInterval - 30s) to repeat the process.

My approach is very hackish and customised, and probably not a good idea to implement in this way as part of the library. At a high level, there should be some way for users to create and customise these functions. Perhaps the architecture can be tied to #3.

**Comment by peoro** [comment]
Yeah, it's really difficult to express similar tasks composed of simpler ones using the current architecture, as discussed in #3.

If you want to add a hacky routine working in the meantime go for it. It would be best to keep the hacky part self-contained: instead of adding `myinterval` and `mytimeout`, we could do what [`killAllLoop`](https://github.com/peoro/ngu.js/blob/master/src/loops.js#L271-L287) does: terminating the running sub-task by throwing an exception at the right moment.

I'd try to find a better way to express complex behaviors soon anyways, as it'd make everything a lot easier.

**Comment by jusw85** [comment]
Yup, never intended to add this function to the library in it's current state, it's too hacky and requires too many workarounds. But I think this is a worthy use case to work toward gradually, starting with #3, then adding OCR, etc. For now I'll stick with my local copy, the current state of the library has reached near feature parity with my branch and is sufficient for me.

**Comment by peoro** [comment]
Yeah... Another thing that I would want is to harvest Ygg. I'd like to have something that stops NGU.js in the middle of whatever its doing to go handle Ygg, as soon as a fruit is ready.

**Comment by peoro** [comment]
I tried to isolate a function to interrupt a running loop in https://github.com/peoro/ngu.js/commit/ccff7efd8b5b896b7a05a8538c4b77d663652561.

It *might* be what we need for quick hacking before switching to a better system.

I have tested it way less than I should have, so play with it for a bit before using it. I'm especially not sure whether it will handle recursion properly (e.g. a loop that `waitTimeout`s a second loop which `waitTimeout`s a third one).
